src/ios/ffi_for_sanity_tests.rs

```diff
@@ -0,0 +1,160 @@
+
+use super::ios_interface::cstring_to_str;
+use core_foundation::{base::TCFType, string::{CFString, CFStringRef}};
+use libc::c_char;
+use std::{thread, sync::mpsc::{self, Sender}};
+use mpsc::Receiver;
+
+// Expose an interface for apps (for now only iOS) to test that general FFI is working as expected.
+// i.e. assumptions on which the actual FFI interface relies.
+// TODO can the c headers for this be generated in a separate file? Needs adjustments in script to generate lib and framework too.
+
+#[repr(C)]
+pub struct FFIParameterStruct {
+  my_int: i32,
+  my_str: *const c_char, // TODO use CFStringRef here too?
+  my_nested: FFINestedParameterStruct,
+}
+
+#[repr(C)]
+pub struct FFINestedParameterStruct {
+  my_u8: u8,
+}
+
+#[derive(Debug)]
+struct MyStruct {
+  my_int: i32,
+  my_str: String,
+  my_u8: u8,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn pass_struct(par: *const FFIParameterStruct) -> i32 {
+  let my_str = cstring_to_str(&(*par).my_str).unwrap();
+
+  let my_struct = MyStruct {
+    my_int: (*par).my_int,
+    my_str: my_str.to_owned(),
+    my_u8: (*par).my_nested.my_u8,
+  };
+
+  println!("Received struct from iOS: {:?}", my_struct);
+
+  1
```

Comment:
@i-schuetz why is it required to return i32? Void not supported?

Reply:
Just returning something to double-check in the unit tests that it's terminating correctly.